src/v/cloud_storage/topic_manifest.cc

```diff
@@ -923,191 +924,6 @@ static auto do_update(DocumentView doc) -> std::
 } // namespace
 namespace cloud_storage {
 
-struct topic_manifest_handler
-  : public json::BaseReaderHandler<json::UTF8<>, topic_manifest_handler> {
-    using key_string = ss::basic_sstring<char, uint32_t, 31>;
-    bool StartObject() {
-        switch (_state) {
-        case state::expect_manifest_start:
```

Comment:
Test suggestion: what if we actually kept the topic_manifest_handler in the tree for some time, but kept it available for tests only. I'm thinking it'd be a great test for backwards compatibility to serialize a random manifest with the old code, and then deserialize the json with the new code, and ensure they match, etc. WDYT?